Add doc to introduce using devfiles on IBM Z & Power. (#4724)
openshift/odo: docs/public/deploying-a-devfile-using-odo-on-IBM-Z-and-Power.adoc

> == Introduction to devfile

Reviewer: I would suggest a title of “using odo on ibm z and power” and this intro being a header 2.

What is a devfile?

> A https://redhat-developer.github.io/devfile/[devfile] is a portable file that describes your development environment. It allows reproducing a _portable_ developmental environment without the need of reconfiguration.

Reviewer 2: ```suggestion
A https://redhat-developer.github.io/devfile/[devfile] is a portable file that describes your development environment. It allows reproducing a _portable_ development environment without the need of reconfiguration.
```

With a devfile you can describe:

* Development components such as container definition for build and application runtimes
* A list of pre-defined commands that can be run
* Projects to initially clone

> odo takes this devfile and transforms it into a workspace of multiple containers running on Kubernetes or OpenShift.

Reviewer 2: ```suggestion
odo takes this devfile and uses it to create a workspace of multiple containers running on Kubernetes or OpenShift.
```

Devfiles are YAML files with a defined https://devfile.github.io/devfile/_attachments/api-reference.html[schema].

== odo and devfile

odo can now create components from devfiles as recorded in registries. odo automatically consults the https://github.com/odo-devfiles/registry[default registry] but users can also add their own registries. Devfiles contribute new component types that users can pull to begin development immediately.

An example deployment scenario:

. `odo create` will consult the recorded devfile registries to offer the user a selection of available component types and pull down the associated `devfile.yaml` file
. `odo push` parses and then deploys the component in the following order:
 .. Parses and validates the YAML file
 .. Deploys the development environment to your Kubernetes or OpenShift cluster
 .. Synchronizes your source code to the containers
 .. Executes any prerequisite commands

== Deploying your first devfile on IBM Z & Power

> Since the DefaultDevfileRegistry don't support IBM Z & Power now, you will need to create one secure private devfileregistry firstly. The steps to create new secure private devfileregistry , please check the doc link:https://github.com/openshift/odo/blob/main/docs/public/secure-registry.adoc[secure registry].

Reviewer 2: ```suggestion
Since the DefaultDevfileRegistry doesn't support IBM Z & Power now, you will need to create a secure private DevfileRegistry first. To create a new secure private DevfileRegistry, please check the doc link:https://github.com/openshift/odo/blob/main/docs/public/secure-registry.adoc[secure registry].
```

[options="header"]
|===
|Language | Devfile Name | Description | Image Source | Supported Platform

| Java
| java-maven
| Upstream Maven and OpenJDK 11
| link:https://registry.redhat.io/codeready-workspaces/plugin-java11-openj9-rhel8[codeready-workspaces/plugin-java11-openj9-rhel8]
| s390x, ppc64le

| Java
| java-openliberty
| Open Liberty microservice in Java
| link:https://registry.redhat.io/codeready-workspaces/plugin-java11-openj9-rhel8[codeready-workspaces/plugin-java11-openj9-rhel8]
| s390x, ppc64le

| Java
| java-quarkus
| Upstream Quarkus with Java+GraalVM
| link:https://registry.redhat.io/codeready-workspaces/plugin-java8-openj9-rhel8[codeready-workspaces/plugin-java8-openj9-rhel8]
| s390x, ppc64le

| Java
| java-springboot
| Spring Boot® using Java
| link:https://registry.redhat.io/codeready-workspaces/plugin-java11-openj9-rhel8[codeready-workspaces/plugin-java11-openj9-rhel8]
| s390x, ppc64le

| Vert.x Java
| java-vertx
| Upstream Vert.x using Java
| link:https://registry.redhat.io/codeready-workspaces/plugin-java11-openj9-rhel8[codeready-workspaces/plugin-java11-openj9-rhel8]
| s390x, ppc64le

| Node.JS
| nodejs
| Stack with NodeJS 12

> | link:https://registry.redhat.io/codeready-workspaces/plugin-java8-openj9-rhel8[codeready-workspaces/plugin-java8-openj9-rhel8]

Reviewer: Clicking these links gives me not found. Which is ok as they are image urls but then they shouldn’t be urls as the first instinct of a user would be click the link.
Maybe point to the dockerfile of the image?

| s390x, ppc64le

| Python
| python
| Python Stack with Python 3.7
| link:https://registry.redhat.io/codeready-workspaces/plugin-java8-openj9-rhel8[codeready-workspaces/plugin-java8-openj9-rhel8]
| s390x, ppc64le

| Django
| python-django
| Python3.7 with Django
| link:https://registry.redhat.io/codeready-workspaces/plugin-java8-openj9-rhel8[codeready-workspaces/plugin-java8-openj9-rhel8]
| s390x, ppc64le

|===
[NOTE]
====
Access to the Red Hat registry is required to use these images on IBM Power Systems & IBM Z.
====

> Steps to use devfiles are link:https://github.com/openshift/odo/blob/main/docs/public/deploying-a-devfile-using-odo.adoc#deploying-your-first-devfile[deploy your first devfile].

Reviewer 2: ```suggestion
Steps to use devfiles can be found in the doc link:https://github.com/openshift/odo/blob/main/docs/public/deploying-a-devfile-using-odo.adoc#deploying-your-first-devfile[deploy your first devfile].
```
